src/idl/tests/annotation.c

```diff
@@ -1387,10 +1389,13 @@ test_rep(
 
   rep_test_t *test = (rep_test_t*)user_data;
 
-  allowable_data_representations_t
-    result = supported_data_representations(node),
+  allowable_data_representations_t result, required;
+    result = idl_supported_data_representations(node),
     required = test->reps[test->i++];
-  CU_ASSERT_EQUAL(required,result);
+  if (idl_requires_xcdr2(node))
+    result &= ~((allowable_data_representations_t)IDL_DATAREPRESENTATION_FLAG_XCDR1);
```

Comment:
This is intriguing: it is almost as if there is a mix-up between "allowed" and "supported". Clearly the old "supported" was changed to "allowed", but then you'd expect that writing `@data_representation(XCDR1|XCDR2)` should result in both being "allowed", regardless of whether `XCDR1` is actually supported for the type.

So I think the test tries mixes two different things: whether or not the type requires `XCDR2` or not, and whether or not the allowed representations match. Wouldn't it then be better to make that explicit in the test expectation?

Reply:
You're right, this is confusing! In [3c9d757](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1179/commits/3c9d75774b3d1addcabf6bc7795edd6f9cbb6725) I've removed the tests for `idl_requires_xcdr2` from the data representation test and added a new test for that. I've also excluded ([ddc4e34](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1179/commits/ddc4e34c7fefe0933d4a1f8a4f9615d2bfb6dbd0)) appendable enums and bitmasks in the requires-xcdr2 check, as there is no reason to force xcdr2 for that